team/bundles/org.eclipse.compare/compare/org/eclipse/compare/contentmergeviewer/TextMergeViewer.java

```diff
@@ -474,6 +475,12 @@ private static RGB interpolate(RGB fg, RGB bg, double scale) {
 	private IUndoableOperation copyUndoable = null;
 	private IOperationHistoryListener operationHistoryListener;
 
+	// https://github.com/eclipse-platform/eclipse.platform.ui/issues/2143
+	// calling w.getLineHeight(line) + w.getLineSpacing() +
+	// w.getLineVerticalIndent(line); again and again for the same line is too slow
+	// we therefore introduce a line height cache by viewer
```

Comment:
Should it be a field, or can it be local variable?

Users can edit & save files in compare editor & so may add/remove lines.
Would this map **as field* mean, the line height will be wrong for the same compare editor after changes?

Reply:
It must be a field, else the recalculation as local variable need too much time. 

Because of textChanges, there is a listener in the method: getHeightBetweenLines which reacts on text changes and clears this field completely.